docarray/doc_index/abstract_doc_index.py

```diff
@@ -626,31 +628,60 @@ def build_query(self) -> QueryBuilder:
         """
         return self.QueryBuilder()  # type: ignore
 
-    def _create_columns(self, schema: Type[BaseDocument]) -> Dict[str, _ColumnInfo]:
-        columns: Dict[str, _ColumnInfo] = dict()
+    def _flatten_schema(
+        self, schema: Type[BaseDocument], name_prefix: str = ''
+    ) -> List[Tuple[str, Type, 'ModelField']]:
+        """Flatten the schema of a Document into a list of column names and types.
```

Comment:
maybe add a line in the docstring about that fact that it handle the nested schema in a recursive manner by adding `__`

Reply:
can we add unit test for this function ?